dozer-types/src/types/mod.rs

```diff
@@ -238,3 +242,103 @@ pub enum Operation {
     Insert { new: Record },
     Update { old: Record, new: Record },
 }
+
+#[derive(Clone, Copy, Debug, Serialize, Deserialize)]
+pub struct DozerCoord(pub Coord);
+
+impl Eq for DozerCoord {}
+
+impl PartialEq for DozerCoord {
+    fn eq(&self, other: &Self) -> bool {
+        self.0.x == other.0.x && self.0.y == other.0.y
+    }
+}
+
+impl Ord for DozerCoord {
+    fn cmp(&self, _other: &Self) -> Ordering {
+        todo!("Cmp for DozerCoord not implemented")
+    }
+}
+
+impl PartialOrd for DozerCoord {
+    fn partial_cmp(&self, _other: &Self) -> Option<Ordering> {
+        todo!("Partial_cmp for DozerCoord not implemented")
+    }
+}
+
+impl Display for DozerCoord {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.write_str(&format!("{:?}", self.0.x_y()))
+    }
+}
+
+impl From<(f64, f64)> for DozerCoord {
+    fn from((x, y): (f64, f64)) -> Self {
+        Self(coord! {x: x, y: y})
+    }
+}
+
+impl DozerCoord {
+    pub fn as_bytes(&self) -> Vec<u8> {
+        let mut x_bytes_vec = self.0.x.encode_to_vec();
+        let mut y_bytes_vec = self.0.y.encode_to_vec();
+        x_bytes_vec.append(&mut y_bytes_vec);
+        x_bytes_vec
+    }
```

Comment:
`as_bytes` usually returns a `&[u8]`. Let's call it `to_bytes`.

We can change it like this

```rust
pub fn to_bytes(&self) -> [u8; 16] {
  let mut result = [0; 16];
  result[0..8].copy_from_slice(self.0.x.to_be_bytes());
  result[8..16].copy_from_slice(self.0.y.to_be_bytes());
  result
}
```

to avoid allocations.